src/subscriber/Broadcaster.ts

```diff
@@ -8,8 +8,15 @@ import { RelationMetadata } from "../metadata/RelationMetadata"
 import { ObjectUtils } from "../util/ObjectUtils"
 
 interface BroadcasterEvents {
-    BeforeQuery: () => void
-    AfterQuery: () => void
+    BeforeQuery: (query: string, parameters: undefined | any[]) => void
+    AfterQuery: (
+        query: string,
+        parameters: undefined | any[],
+        success: boolean,
+        executionTime: undefined | number,
+        rawResults: undefined | any,
+        error: undefined | any,
+    ) => void
```

Comment:
For readability prefer to write `undefined` last in type unions, so the expected type is first:

```suggestion
    BeforeQuery: (query: string, parameters: any[] | undefined) => void
    AfterQuery: (
        query: string,
        parameters: any[] | undefined,
        success: boolean,
        executionTime: number | undefined,
        rawResults: any | undefined,
        error: any | undefined,
    ) => void
```